platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpHandler.java

```diff
@@ -606,12 +604,23 @@ private String encodeRedirectRequest(String request) throws WSSecurityException,
     return URLEncoder.encode(RestSecurity.deflateAndBase64Encode(request), "UTF-8");
   }
 
-  private String encodePostRequest(String request) throws WSSecurityException, IOException {
+  private String encodePostRequest(String request) throws WSSecurityException {
     return Base64.getEncoder().encodeToString(request.getBytes(StandardCharsets.UTF_8));
   }
 
   private String recreateFullRequestUrl(HttpServletRequest request) {
     StringBuffer requestURL = request.getRequestURL();
+
+    try {
+      URL url = new URL(requestURL.toString());
+      if (url.getHost().equals(SystemBaseUrl.EXTERNAL.getHost())
+          && String.valueOf(url.getPort()).equals(SystemBaseUrl.EXTERNAL.getPort())) {
+        requestURL = new StringBuffer(SystemBaseUrl.EXTERNAL.constructUrl(request.getRequestURI()));
```

Comment:
❓ Is this effectively adding the external context? Is it guaranteed that the external context will not be returned from `request.getRequestURI()` already? (otherwise, the external context will be in there twice).

Reply:
I'll add a check for that just in case